recordtbl_view.php

```diff
@@ -0,0 +1,432 @@
+						{
+							if ( $fieldName == $recordIDField ||
+							     $fieldName == 'redcap_data_access_group' )
+							{
+								$columnName = $fieldName;
+								$listColumns[ $columnName ] = true;
+								$resultRow[ $columnName ] = $fieldValue;
+							}
+							elseif ( is_array( $fieldValue ) )
+							{
+								$fieldOptions = $fieldValue;
+								foreach ( $fieldOptions as $fieldOption => $fieldValue )
+								{
+									$columnName = ( $showEventNames ? ( $eventName . '__' ) : '' ) .
+									              $fieldName . '__' . $instanceID . '___' .
+									              $fieldOption;
+									$listColumns[ $columnName ] = true;
+									$resultRow[ $columnName ] = $fieldValue;
+								}
+							}
+							else
+							{
+								$columnName = ( $showEventNames ? ( $eventName . '__' ) : '' ) .
+								              $fieldName . '__' . $instanceID;
+								$listColumns[ $columnName ] = true;
+								if ( $fieldValue != '' )
+								{
+									if ( in_array( $fieldName, $listDateFields ) )
+									{
+										$resultRow[ $columnName ] =
+											\DateTimeRC::format_ts_from_ymd( $fieldValue,
+											                                 false, true );
+									}
+									elseif ( isset( $listOptionFields[ $fieldName ] ) )
+									{
+										$resultRow[ $columnName ] =
+											$listOptionFields[ $fieldName ][ $fieldValue ] .
+											' (' . $fieldValue . ')';
+									}
+									else
+									{
+										$resultRow[ $columnName ] = $fieldValue;
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+			continue;
+		}
+		$eventName = $listEvents[ $eventID ];
+		foreach ( $listFields as $fieldName => $fieldValue )
+		{
+			if ( $fieldName == $recordIDField || $fieldName == 'redcap_data_access_group' )
+			{
+				$columnName = $fieldName;
+				$listColumns[ $columnName ] = true;
+				$resultRow[ $columnName ] = $fieldValue;
+			}
+			elseif ( is_array( $fieldValue ) )
+			{
+				$fieldOptions = $fieldValue;
+				foreach ( $fieldOptions as $fieldOption => $fieldValue )
+				{
+					$columnName = ( $showEventNames ? ( $eventName . '__' ) : '' ) .
+					              $fieldName . '__1___' . $fieldOption;
+					$listColumns[ $columnName ] = true;
+					$resultRow[ $columnName ] = $fieldValue;
+				}
+			}
+			else
+			{
+				$columnName = ( $showEventNames ? ( $eventName . '__' ) : '' ) . $fieldName . '__1';
+				$listColumns[ $columnName ] = true;
+				if ( $fieldValue != '' )
+				{
+					if ( in_array( $fieldName, $listDateFields ) )
+					{
+						$resultRow[ $columnName ] =
+							\DateTimeRC::format_ts_from_ymd( $fieldValue, false, true );
+					}
+					elseif ( isset( $listOptionFields[ $fieldName ] ) )
+					{
+						$resultRow[ $columnName ] = $listOptionFields[ $fieldName ][ $fieldValue ] .
+						                            ' (' . $fieldValue . ')';
+					}
+					else
+					{
+						$resultRow[ $columnName ] = $fieldValue;
+					}
+				}
```

Comment:
Looks similar to line 174 onwards? Might be able to extract into a function to avoid duplication.